scvi/dataloaders/_data_splitting.py

```diff
@@ -516,3 +517,149 @@ def __getitem__(self, idx: List[int]) -> Dict[str, torch.Tensor]:
     def __len__(self):
         for _, value in self.data.items():
             return len(value)
+
+
+class ContrastiveDataSplitter(DataSplitter):
```

Comment:
Ideally this code should also be under `scvi.external.contrastivevi`.

Reply:
Okay, I moved it under `scvi.external.contrastivevi`. The test cases are also now under `tests/external`.